### PR TITLE
add mailroom3

### PR DESCRIPTION
**Pull request by srmorehouse**

### CODE REVIEW

**Comment by natasha-aleksandrova** [comment]
@charlesrobison please grade according to the work submitted - you had the same comment about dictionary in the previous PR. Thanks.

**Comment by srmorehouse** [comment]
updated mailroom2, 3, 4 to use dictionary and sort correctly

File: students/srmorehouse/Lesson05/mailroom3.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        # now redirect to feature functions based on the user selection
+        if name == "list":
+            list_donors()
+        elif not any(name in i for i in donor_db):
+            donor_db.update({name: []})
+            break
+        else:
+            for i in range (len (donor_db)):
+                if donor_db[i][0] == name:
+                    break
```
Comment (charlesrobison) [inline students/srmorehouse/Lesson05/mailroom3.py]: Getting an error when trying to add a new donation to an existing donor.

File: students/srmorehouse/Lesson06/mailroom4.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                if donor_db[i][0] == name:
+                    break
+
+    # enter donation amount
+    donation = float(input("Enter a donation amount: "))
+    donor_db[name].append(donation)
+
+    # write email
+    msg = f"\n{name},\n\nThank you for your donation of ${donation}.\n"
+    print(msg)
```
Comment (charlesrobison) [inline students/srmorehouse/Lesson06/mailroom4.py]: This is having the same issue as your mailroom2 file, isn't allowing me to add new donations for donors - it's overwriting each donation.

**Comment by charlesrobison** [comment]
@srmorehouse - not sure if you were ready to have Lesson 6 mailroom reviewed - when complete please submit on the course site for grading.

**Pull request merged**